py/functions.py

```diff
@@ -0,0 +1,212 @@
+
+    Args:
+        files: List of file paths.
+        targetHash: Hash value to compare with.
+
+    Returns:
+        Union[str, None]: Path of the file if found, None otherwise.
+    """
+    for file in files:
+        if not isImg(file):
+            continue
+        fileHash = genHash(file)
+        if fileHash == targetHash:
+            return file
+    return None
+    # File paths can be stored in DB but what if path is changed?
+    # we need to keep checking for the path change and update DB (TBI)
+
+
+def processImgs(conn: sqlite3.Connection, files: List[str]) -> None:
+    """
+    Process images and insert data into the database.
+
+    Args:
+        conn: SQLite database connection.
+        files: List of file paths.
+    """
+    for file in files:
+        if not isImg(file):
+            continue
+        imgHash = genHash(file)
+        if hashExist(conn, imgHash):
+            continue
+        imgClass = classifyImg(file)
+        query = f"INSERT OR REPLACE INTO media(hash, imageClass) VALUES('{imgHash}', '{imgClass}')"
+        executeQuery(conn, query)
+
+
+def connectDB(dbPath: str) -> sqlite3.Connection:
+    """
+    Connect to an SQLite database.
+
+    Args:
+        dbPath: Path to the database file.
+
+    Returns:
+        sqlite3.Connection: Database connection.
+    """
+    conn = sqlite3.connect(dbPath)
+    return conn
+
+
+def createTable(conn: sqlite3.Connection, tableID: str, columns: List[str]) -> None:
+    """
+    Create a table in the database if it doesn't exist.
+
+    Args:
+        conn: SQLite database connection.
+        tableID: Name of the table.
+        columns: List of column names and types.
+    """
+    cursor = conn.cursor()
+    cursor.execute(f"CREATE TABLE IF NOT EXISTS {tableID} ({', '.join(columns)})")
+    conn.commit()
+
+
+def executeQuery(conn: sqlite3.Connection, query: str) -> List[Tuple]:
+    """
+    Execute a SQL query and return the results.
+
+    Args:
+        conn: SQLite database connection.
+        query: SQL query to execute.
+
+    Returns:
+        List[Tuple]: Results of the query.
+    """
+    cursor = conn.cursor()
+    cursor.execute(query)
+    return cursor.fetchall()
+
+
+def closeConnection(conn: sqlite3.Connection) -> None:
+    """
+    Close the database connection.
+
+    Args:
+        conn: SQLite database connection.
+    """
+    conn.commit()
+    conn.close()
+
+
+def hashExist(conn: sqlite3.Connection, hashValue: str) -> bool:
+    """
+    Check if a hash value exists in the database.
+
+    Args:
+        conn: SQLite database connection.
+        hashValue: Hash value to check.
+
+    Returns:
+        bool: True if the hash value exists, False otherwise.
+    """
+    query = f"SELECT EXISTS(SELECT 1 FROM media WHERE hash='{hashValue}')"
+    result = executeQuery(conn, query)
+    return result[0][0]
+
+
+def fileByClass(conn: sqlite3.Connection, files: List[str], tableID: str) -> Dict[str, List[str]]:
```

Comment:
This function wouldn't be required if we could find an effective way to store paths of images in the database. Then, we could simply run a query to group paths according to a unique set of classes, making it more efficient than checking hashes of each file.